pkg/nimble/netif/nimble_netif.c

```diff
@@ -0,0 +1,621 @@
+/*
+ * Copyright (C) 2018-2019 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     pkg_nimble_gnrc
+ *
+ * # TODO
+ * - many many things...
+ *
+ *
+ * @{
+ *
+ * @file
+ * @brief       GNRC netif wrapper for NimBLE
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ *
+ * @}
+ */
+
+#include <limits.h>
+
+#include "assert.h"
+#include "thread.h"
+#include "thread_flags.h"
+
+#include "net/ble.h"
+#include "net/bluetil/addr.h"
+#include "net/gnrc/netif.h"
+#include "net/gnrc/netif/hdr.h"
+#include "net/gnrc/netreg.h"
+#include "net/gnrc/pktbuf.h"
+#include "net/gnrc/nettype.h"
+
+#include "nimble_netif.h"
+#include "nimble_netif_conn.h"
+#include "nimble_riot.h"
+#include "host/ble_gap.h"
+#include "host/util/util.h"
+
+#define ENABLE_DEBUG            (0)
+#include "debug.h"
+
+#ifdef MODULE_GNRC_SIXLOWPAN
+#define NETTYPE                 GNRC_NETTYPE_SIXLOWPAN
+#else
+#define NETTYPE                 GNRC_NETTYPE_UNDEF
```

Comment:
As you use this to dispatch to the upper layer directly, shouldn't there also a fall-back to IPv6 if it is available but 6Lo is not?

Reply:
sure, doesn't hurt